### PR TITLE
Midna CLI builds working

### PR DESCRIPTION
**Pull request by NikitaVr**
- changed a bunch of the dependencies and typescript configs to get built typescript working
- Building and running should work for all 3 now
  - Hardhat example deploy
  - Web app
  - Midna CLI

## Testing

Update all dependencies

```
yarn
```

### Run the web app

```
yarn web:dev
```

### Hardhat test

Update .env inside hardhat-example

```
MIDNA_URL=http://localhost:3000
MIDNA_API_URL=http://localhost:3000/api
```

start hardhate node

```
cd hardhat-example 
npx hardhat node
```

Deploy hardhat script (from new terminal)

```
cd hardhat-example 
npx hardhat run --network localhost scripts/deploy.ts
```

### CLI Test

Update .env inside midna-cli

```
MIDNA_URL=http://localhost:3000
MIDNA_API_URL=http://localhost:3000/api
```

Run CLI Command

```
cd midna-cli
./bin/run connect Counter 31337 0x34871293 0x53243 12312
```

## Screenshots
image

### CODE REVIEW

File: hardhat-example/package.json [hunk 1]
```diff
@@ -5,17 +5,19 @@
     "hardhat": "^2.17.3"
   },
   "dependencies": {
-    "@midna/sdk": "workspace:*",
+    "@midna/sdk": "/Users/nikita/Documents/chainpatrol/midna-sdk",
```
Comment (umar-ahmed) [inline hardhat-example/package.json]: This won't work on other's machines

File: hardhat-example/package.json [hunk 2]
```diff
@@ -5,17 +5,19 @@
...
     "@nomicfoundation/hardhat-chai-matchers": "^2.0.0",
     "@nomicfoundation/hardhat-ethers": "^3.0.0",
     "@nomicfoundation/hardhat-network-helpers": "^1.0.0",
     "@nomicfoundation/hardhat-verify": "^1.0.0",
     "@typechain/ethers-v6": "^0.4.0",
     "@typechain/hardhat": "^8.0.0",
+    "add": "^2.0.6",
```
Comment (umar-ahmed) [inline hardhat-example/package.json]: where is this package used?

File: hardhat-example/package.json [hunk 3]
```diff
@@ -5,17 +5,19 @@
...
     "dotenv": "^16.3.1",
     "ethers": "^6.4.0",
     "hardhat-gas-reporter": "^1.0.8",
     "solidity-coverage": "^0.8.1",
-    "typechain": "^8.2.0"
+    "typechain": "^8.2.0",
+    "yarn": "^1.22.19"
```
Comment (umar-ahmed) [inline hardhat-example/package.json]: yarn shouldn't be added as a dependency

File: packages/db/.gitattributes [hunk 4]
```diff
@@ -0,0 +1,2 @@
+# Auto detect text files and perform LF normalization
+* text=auto
```
Comment (umar-ahmed) [inline packages/db/.gitattributes]: Why is this necessary?

File: packages/rest/.gitignore
Comment (umar-ahmed) [inline packages/rest/.gitignore:1]: A lot of the stuff in this gitignore doesn't need to be repeated in each package. Just add to a root gitignore

File: packages/rest/package.json [hunk 5]
```diff
@@ -1,13 +1,21 @@
 {
   "name": "@midna/rest",
-  "private": true,
-  "main": "./src/index.ts",
-  "types": "./src/index.ts",
-  "peerDependencies": {
-    "zod": "^3.22.2"
+  "version": "1.0.0",
+  "description": "Can log \"hello world\" and \"goodbye world\" to the console!",
```
Comment (umar-ahmed) [inline packages/rest/package.json]: ?

File: packages/rest/package.json [hunk 6]
```diff
@@ -1,13 +1,21 @@
 {
   "name": "@midna/rest",
-  "private": true,
```
Comment (umar-ahmed) [inline packages/rest/package.json:3]: Package should remain private

File: web/data.db
Comment (umar-ahmed) [inline web/data.db:1]: should be gitignored

File: packages/utils/package.json [hunk 7]
```diff
@@ -1,12 +1,19 @@
...
+  },
+  "files": [
+    "/dist"
+  ],
   "dependencies": {
-    "@types/git-rev-sync": "^2.0.0",
-    "@types/github-url-from-git": "^1.5.1",
-    "git-rev-sync": "^3.0.2",
-    "github-url-from-git": "^1.5.0"
+    "tslib": "^2.6.2"
```
Comment (umar-ahmed) [inline packages/utils/package.json]: tslib seems unecessary, I think it's only used if `importHelpers` are enabled

File: packages/sdk/src/index.ts [hunk 8]
```diff
@@ -5,7 +5,7 @@ import * as fs from "node:fs";
 import * as ethers from "ethers";
 import { createClient } from "@midna/rest";
 import { getChainName } from "@midna/utils";
-import { getCommitLink } from "./git";
+import { getCommitLink } from "./git.js";
```
Comment (umar-ahmed) [inline packages/sdk/src/index.ts]: why the file extension here?

File: packages/db/package.json [hunk 9]
```diff
@@ -1,22 +1,26 @@
...
-  },
-  "peerDependencies": {
-    "zod": "^3.22.2"
+    "build": "tsc"
   },
+  "files": [
+    "/dist"
+  ],
   "dependencies": {
+    "@types/better-sqlite3": "^7.6.5",
```
Comment (umar-ahmed) [inline packages/db/package.json]: dev dep

File: packages/db/package.json [hunk 10]
```diff
@@ -1,22 +1,26 @@
 {
   "name": "@midna/db",
-  "private": true,
-  "main": "./src/index.ts",
-  "types": "./src/index.ts",
+  "version": "1.0.0",
+  "description": "Can log \"hello world\" and \"goodbye world\" to the console!",
```
Comment (umar-ahmed) [inline packages/db/package.json]: ?

**Review by umar-ahmed — CHANGES_REQUESTED** [review]
There's some references to your local filesystem still. Also, there are some unnecessary files and modules included

File: packages/db/.gitattributes (hunk 4 above)
Comment (NikitaVr) [reply packages/db/.gitattributes]: some copy pasta, removing

File: packages/rest/.gitignore
Comment (NikitaVr) [reply packages/rest/.gitignore:1]: good point, moved it up

File: packages/rest/package.json (hunk 5 above)
Comment (NikitaVr) [reply packages/rest/package.json]: copy pasta, fixed

File: packages/db/package.json (hunk 10 above)
Comment (NikitaVr) [reply packages/db/package.json]: copy paste, cleaned up

File: .gitignore [hunk 11]
```diff
@@ -21,3 +21,35 @@ output.json
...
+
+# OS metadata
+.DS_Store
+Thumbs.db
+
+# Ignore built ts files
+/**/dist/**/*
+
+# ignore yarn.lock
+yarn.lock
```
Comment (umar-ahmed) [inline .gitignore]: should not ignore lockfile, it's what determines the actual packages versions to pull in

File: packages/db/package.json [hunk 12]
```diff
@@ -1,22 +1,26 @@
...
+  "files": [
+    "/dist"
+  ],
   "dependencies": {
     "better-sqlite3": "^8.6.0",
     "dotenv": "^16.3.1",
     "drizzle-orm": "^0.28.6",
-    "drizzle-zod": "^0.5.1"
+    "drizzle-zod": "^0.5.1",
+    "tslib": "^2.6.2",
```
Comment (umar-ahmed) [inline packages/db/package.json]: This should be removed from all packages

File: packages/rest/package.json [hunk 13]
```diff
@@ -1,13 +1,21 @@
...
   },
+  "files": [
+    "/dist"
+  ],
   "dependencies": {
-    "@midna/db": "workspace:*",
-    "@ts-rest/core": "^3.30.2"
+    "@ts-rest/core": "^3.30.2",
+    "tslib": "^2.6.2",
+    "zod": "^3.22.2"
```
Comment (umar-ahmed) [inline packages/rest/package.json:14]: zod can be listed as a peer dependency in all these packages. Many applications use it themselves, so describing it here might pull in duplicate versions

**Review by umar-ahmed — CHANGES_REQUESTED** [review]
still see some things that should be removed

**Review by umar-ahmed — APPROVED** [review]
lgtm

**Pull request merged**